### ISSUE TITLE
Training Time

### ISSUE THREAD

**Issue by chang0517** [dome272/Diffusion-Models-pytorch#2]
Hi. I'm the nuvie studying about Diffusion models.
First, thank you for interesting video and code.

I have some question about the code and implementation.

1. Is conditional image generation possible for 32x32 CIFAR-10 Images? Then, is it possible to adjust the Unet structure and various parameters? As a result of training and inference, the image is not well generated. If there is a reason, I would like to know what it is.

2. Can you tell me how long the train time is when you train on 64x64?

Thank you.

**Comment by dome272** [comment]
Hey there,
thanks for the nice comment.

1. In order to help you better I would need more information on your training run. What exactly did you change in my code? Did you just change the path to the CIFAR-10 dataset at 32x32? If so you also need to change a few things in the code. I assume you changed the things already to make the training code run: [Changing dataloader resizing](https://github.com/dome272/Diffusion-Models-pytorch/blob/main/utils.py#L26), [train image size](https://github.com/dome272/Diffusion-Models-pytorch/blob/main/ddpm_conditional.py#L119), [modify the inputs to the self-attention blocks](https://github.com/dome272/Diffusion-Models-pytorch/blob/main/modules.py#L189). Can you confirm that you changed all these things? If so, I would need more information on how your sampled images looked like, for how long you trained and how the MSE metric looks like (its being logged to tensorboard so you can just open it like this).

2. I trained for 300 epochs on the 64x64 datasets (CIFAR-10 and Landscape), which took around 3 days on an rtx 3090.

**Comment by karkay22** [comment]
Hi, thank for your video which was really interesting and well explained.

I'm sorry, I'm new to diffusion models and I'm trying to train your model with CFG on an MNIST dataset like this one : https://www.kaggle.com/datasets/scolianni/mnistasjpg or  on CIFAR10-32

I encounter some troubles, I changed the dataloader resizing to 40, the image size to 32 but got an error : shape '[-1, 128, 32, 32]' is invalid for input of size 458752.

Also, I m'not sure of what I have to change in the attention-blocks, tried multiple things for the inputs of each layer but can't find the good structure, could you tell me what I have to change and the values I should put instead ?

Hope you'll be able to help me.

Thank you
